scene/gui/graph_edit.cpp

```diff
@@ -1646,6 +1646,585 @@ HBoxContainer *GraphEdit::get_zoom_hbox() {
+			}
+		}*/
+
+		//Minimise crossings
+		l.set(i, split(lower_layer, c));
+	}
+
+	//Print layers here
+	//Check if vectors of hashmap are being written
+	for (int i = 0; l.has(i); i++) {
+		for (int j = 0; j < l[i].size(); j++) {
+			StringName temp = l[i][j];
+		}
+	}
+}
+
+void GraphEdit::calculate_inner_shifts(Dictionary &d, const Dictionary &root, const Dictionary &node_names, const Dictionary &align, const Set<StringName> &block_heads, const HashMap<StringName, Pair<int, int>> &port_info) {
+	for (const Set<StringName>::Element *E = block_heads.front(); E; E = E->next()) {
+		real_t left = 0;
+		StringName u = E->get();
+		StringName v = align[u];
+		while (u != v && (StringName)root[u] != v) {
+			String _connection = String(u) + " " + String(v);
+			GraphNode *gfrom = Object::cast_to<GraphNode>(node_names[u]);
+			GraphNode *gto = Object::cast_to<GraphNode>(node_names[v]);
+
+			Pair<int, int> ports = port_info[_connection];
+			int pfrom = ports.first;
+			int pto = ports.second;
+			Vector2 frompos = gfrom->get_connection_output_position(pfrom);
+			Vector2 topos = gto->get_connection_input_position(pto);
+
+			real_t s = (real_t)d[u] + frompos.y - topos.y;
+			d[v] = s;
+			left = left < s ? left : s;
+
+			u = v;
+			v = (StringName)align[v];
+		}
+
+		u = E->get();
+		do {
+			d[u] = (real_t)d[u] - left;
+			u = (StringName)align[u];
+		} while (u != E->get());
+	}
+	//Test for inner_shifts
+	/*List<Variant> _keys;
+	d.get_key_list(&_keys);
+	for (const List<Variant>::Element *E = _keys.front(); E; E = E->next()) {
+		StringName n = (StringName)E->get();
+		real_t inner_shift_val = d[n];
+	}*/
+}
+
+float GraphEdit::calculate_threshold(StringName v, StringName w, const Dictionary &node_names, const HashMap<int, Vector<StringName>> &_layers, const Dictionary &root, const Dictionary &align, const Dictionary &inner_shift, real_t current_threshold, const HashMap<StringName, Vector2> &node_positions, List<Pair<StringName, Connection>> &pending_blocks) {
+#define INIT_ORDER 0x7FFFFFFF
+#define ORDER(_node, _layers)                           \
+	for (unsigned int i = 0; i < _layers.size(); i++) { \
+		int index = _layers[i].find(_node);             \
+		if (index > 0) {                                \
+			order = index;                              \
+			break;                                      \
+		}                                               \
+		order = -1;                                     \
+	}
+
+	int order = -1;
+	float threshold = current_threshold;
+	if (v == w) {
+		int min_order = INIT_ORDER;
+		Connection incoming;
+		for (List<Connection>::Element *E = connections.front(); E; E = E->next()) {
+			//Pick incoming connections
+			if (E->get().to == w) {
+				ORDER(E->get().from, _layers);
+				if (min_order > order) {
+					min_order = order;
+					incoming = E->get();
+				}
+			}
+		}
+
+		if (incoming.from != StringName()) {
+			GraphNode *gfrom = Object::cast_to<GraphNode>(node_names[incoming.from]);
```

Comment:
Godot uses `snake_case`. You are using `gfrom` as your variable names. Please change style.

Reply:
It's completely fine to name local variables like that. It can even be `gf` or something.